pkg/service/catalog/api/types.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+type CatalogService struct {
+	// Target is a reference to a service
+	Target kapi.ObjectReference
+	// Type is a published service's claim type
+	Type string
+	// Description is a user-defined description of the published service
+	Description string
```

Comment:
Yeah, I was expecting a set of export annotations, that defined labels,
description, display names, and probably arbitrary annotations.  We can't
automatically expose every annotation on the root service I don't think.

On Oct 2, 2015, at 3:12 PM, Andy Goldstein <notifications@github.com> wrote:

In pkg/service/catalog/api/types.go
<https://github.com/openshift/origin/pull/4822#discussion_r41057015>:

> @@ -0,0 +1,21 @@
> +package api
> +
> +import (
> +	kapi "k8s.io/kubernetes/pkg/api"
> +)
> +
> +type CatalogService struct {
> +	// Target is a reference to a service
> +	Target kapi.ObjectReference
> +	// Type is a published service's claim type
> +	Type string
> +	// Description is a user-defined description of the published service
> +	Description string

The description for the service in the namespace is "something I exported
for my team" and the description in the catalog is a multi-paragraph intro
to the service including usage info. Seems pretty different. Exported
services are targeted at consumers. Services in the UI are targeted at
namespace owners.

How do you envision being able to set the description on the catalog entry?
2 different annotations on the original service?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4822/files#r41057015>.